libs/cdk/src/lib/components/pagination/services/pagination-head.service.ts

```diff
@@ -29,7 +28,12 @@ export class PaginationHeadService implements OnDestroy {
   }
 
   ngOnDestroy(): void {
-    this._metaService.removeTag('name="canonical"');
+    const element = this._document.querySelector(`link[rel='canonical']`) || null;
```

Comment:
This and the following line could probably be moved into a own private method to reduce code duplication